interface/patient_file/deleter.php

```diff
@@ -461,37 +461,53 @@ function popup_close() {
-                <span class="font-weight-bold">
-                    <?php
-                    if ($patient) {
-                        echo xlt('patient') . " " . text($patient);
-                    } elseif ($encounterid) {
-                        echo xlt('encounter') . " " . text($encounterid);
-                    } elseif ($formid) {
-                        echo xlt('form') . " " . text($formid);
-                    } elseif ($issue) {
-                        echo xlt('issue') . " " . text($issue);
-                    } elseif ($document) {
-                        echo xlt('document') . " " . text($document);
-                    } elseif ($payment) {
-                        echo xlt('payment') . " " . text($payment);
-                    } elseif ($billing) {
-                        echo xlt('invoice') . " " . text($billing);
-                    } elseif ($transaction) {
-                        echo xlt('transaction') . " " . text($transaction);
-                    }
-                    ?>
-                </span>
-                <?php echo xlt('and all subordinate data? This action will be logged'); ?>!
+        <form method='post' name="deletefrm" action='deleter.php?patient=<?php echo attr_url($patient) ?>&encounterid=<?php echo attr_url($encounterid) ?>&formid=<?php echo attr_url($formid) ?>&issue=<?php echo attr_url($issue) ?>&document=<?php echo attr_url($document) ?>&payment=<?php echo attr_url($payment) ?>&billing=<?php echo attr_url($billing) ?>&transaction=<?php echo attr_url($transaction); ?>&csrf_token_form=<?php echo attr_url(CsrfUtils::collectCsrfToken()); ?>'>
+            <input type="hidden" name="csrf_token_form"
+                value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+            <p>
+            <?php
+            $type = '';
+            $id = '';
+            if ($patient) {
+                $id = $patient;
+                $type = 'patient';
+            } elseif ($encounterid) {
+                $id = $encounterid;
+                $type = 'encounter';
+            } elseif ($formid) {
+                $id = $formid;
+                $type = 'form';
+            } elseif ($issue) {
+                $id = $issue;
+                $type = 'issue';
+            } elseif ($document) {
+                $id = $document;
+                $type = 'document';
+            } elseif ($payment) {
+                $id = $payment;
+                $type = 'payment';
+            } elseif ($billing) {
+                $id = $billing;
+                $type = 'invoice';
+            } elseif ($transaction) {
+                $id = $transaction;
+                $type = 'transaction';
+            }
+
+            $ids = explode(",", $id);
+            if (count($ids) > 1) {
+                $type .= 's';
+            }
+
+            $msg = 'You have selected ' . count($ids) . ' ' . $type . " to delete. Are you sure you want to continue?";
+            echo xlt($msg);
```

Comment:
hi @tpantelis ,
If only our translation engine were smart enough to do this. Note the xl function (and related xlt/xla/xlj) take a string and see if it exists in translation tables and then uses translation if it exists. The problem with doing this is that it will never match to anything. We also rely on the xl() family of functions to find the constants that need to be placed into the translation spreadsheet, so much better to place the string inside xl function instead of a variable
Recommend:
1. surrounding each $type setting with xl() function:
```
$type = xl('transaction');
```
2. make the message something like:
```
$msg = xl("You have selected to delete") . ' ' . count($ids) . ' ' . $type . ". " . xl("Are you sure you want to continue?");
echo text($msg);
```

Reply:
Sorry I missed the usage and functionality of `xlt` et al.  Thanks the detailed explanation and suggestions.

Reply:
I've been around 6 years and I still mess up translations.:) or i'm especially bad at escaping. That's why ole eagle eye(I call em) Brady is so helpful!

Reply:
Actually Thomas, I don't think xl() escapes like xlt/xla/xlj does and since I think $msg ends up to user via js ajax(going by memory so, it's old and could be wrong), i'd escape $msg.
e.g js_escape($msg) (really turns out as json_escape($msg))

I'll go ahead and merge once this is done. I want to see about getting rid of the additional confirmation alert.

Reply:
OK - I'll make it `echo text(js_escape($msg));`.  

yeah I don't think the confirmation/information popup alert re: success is necessary. We want to alert on failure but that would appear in the original Yes/No prompt window. I was going to look into that when I get some cycles but feel free.

Reply:
Oh BTW I was formal on my GitHub account but you can call me Tom 😄

Reply:
Tom it is!
Yep, text() is better. This is why I rarely jump in on the escaping stuff(my brain is too meager for such trivial:))

No worries on extra confirm. I'm going to turn the whole affair into a promised dialog.

Reply:
Ya don't need the js_escape() if you're using text(). Same result in the end!

Reply:
Done